[export][refactor][2/n] Move tracing logic

angelayi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114787
* __->__ #114768
* #114764

2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

Differential Revision: [D51823961](https://our.internmc.facebook.com/intern/diff/D51823961)

angelayi committed:
[export][refactor][2/n] Move tracing logic

2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][2/n] Move tracing logic"


2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][2/n] Move tracing logic"


2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][2/n] Move tracing logic"


2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][2/n] Move tracing logic"


2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

[ghstack-poisoned]

angelayi committed:
Update on "[export][refactor][2/n] Move tracing logic"


2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

[ghstack-poisoned]

ydwu4 commented on torch/_export/__init__.py:
```diff
@@ -229,90 +219,6 @@ def _eval(self, mode: bool = True):
-                    for key, (path, ty) in nn_module_stack.items()
-                }
-
 def _export_to_torch_ir(
```
why we want to keep this function and reforward to _trace._export_to_torch_ir? Can we just do from torch.export._trace import _export_to_torch_ir and remove this forwarding?

angelayi replied:
Eventually I want to get rid of reliance on torch/_export, and move all calls to torch.export._trace.export_to_torch_ir. Might as well remove it in this diff and do the forwarding in torch._export. Does that make sense?

ydwu4 commented on torch/_export/__init__.py:
```diff
@@ -588,262 +381,17 @@ def _export(
-
+    from torch.export._trace import _export
 
-    gm_torch_level = _export_to_torch_ir(
```
Maybe also remove this forwarding if we can.

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51823961).

angelayi committed:
Update on "[export][refactor][2/n] Move tracing logic"


2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

Differential Revision: [D51823961](https://our.internmc.facebook.com/intern/diff/D51823961)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51823961).
